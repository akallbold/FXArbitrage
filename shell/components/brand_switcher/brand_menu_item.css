@import 'playa-styleguide/src/stylesheets/variables.css';

.item {
  padding: 5px 8px 1px;
}

.heading {
  margin-top: 15px;
  padding-top: 15px;
  border-top: solid 1px var(--grey);
}

.heading:first-child {
  margin-top: 0;
  border: none;
}

.item:not(.heading) {
  padding: 0 5px 0 26px;
  &:hover {
    background-color: #eaeaea;
  }
}

.link {
  display: inline-block;
  width: 100%;
  padding: 8px 0;
  font-family: var(--font-family-semibold);
  color: var(--text-grey-dark);
  font-size: 14px;
  text-decoration: none;
}

.selected .link {
  color: var(--hyfn-blue);
}

.heading .link {
  color: var(--text-grey-light);
  pointer-events: none;
}

.keyedTo {
  background: var(--background-grey-dark);
}