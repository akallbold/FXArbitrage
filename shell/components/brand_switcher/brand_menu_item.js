import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'redux-little-router'
import { oneLine } from 'common-tags'
import Styles from './brand_menu_item.css'

const classNames = ({ classes, isSelected, isHeading, arrowKeyedTo }) => (
  oneLine`
    ${Styles.item}
    ${classes}
    ${isSelected ? Styles.selected : ''}
    ${isHeading ? Styles.heading : ''}
    ${(arrowKeyedTo && !isHeading) ? Styles.keyedTo : ''}`
)

export const BrandMenuItem = ({
  label,
  hrefPathname,
  classes,
  isSelected,
  isHeading,
  arrowKeyedTo,
  itemRef,
  orgId
}) => (
  <div
    ref={itemRef}
    className={classNames({ classes, isSelected, isHeading, arrowKeyedTo })}
  >
    <Link
      href={{
        pathname: hrefPathname,
        query: { refresh: orgId }
      }}
      className={Styles.link}
    >{label}</Link>
  </div>
)

BrandMenuItem.propTypes = {
  label: PropTypes.string,
  hrefPathname: PropTypes.string,
  classes: PropTypes.string,
  isSelected: PropTypes.bool,
  isHeading: PropTypes.bool
}

BrandMenuItem.defaultProps = {
  label: 'Brand Name Here',
  hrefPathname: '/brands/[someBrandId]/[channel]/manager/',
  classes: '',
  isSelected: false,
  isHeading: false
}

export default BrandMenuItem
