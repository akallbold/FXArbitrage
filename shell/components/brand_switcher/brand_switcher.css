@import 'playa-styleguide/src/stylesheets/variables.css';

.switcher {
  display: inline-flex;
  position: relative;
  font-family: var(--font-family-regular);
  font-size: 12px;
  font-weight: 600;
  max-width: 300px;
  transition: all 1s ease-out;
}

:global(.no-content).switcher {
  max-width: 50px;
}

.button {
  cursor: pointer;
}

.secondaryText {
  margin-bottom: 5px;
  color: rgba(255, 255, 255, 0.5);
}

.primaryText {
  white-space: nowrap;
  color: var(--white);
  font-size: 14px;
}

.downArrowIcon {
  display: inline-block;
  fill: var(--white);
  width: 10px;
  margin: 0 0 0 9px;
}

.tinySpinner {
  width: 14px;
  height: 14px;
  opacity: 0.7;
  margin-top: -3px;
}
