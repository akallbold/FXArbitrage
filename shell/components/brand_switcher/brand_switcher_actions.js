export const OPEN_BRAND_SWITCHER = 'OPEN_BRAND_SWITCHER'
export const CLOSE_BRAND_SWITCHER = 'CLOSE_BRAND_SWITCHER'

export const openBrandSwitcher = () => ({ type: OPEN_BRAND_SWITCHER })
export const closeBrandSwitcher = () => ({ type: CLOSE_BRAND_SWITCHER })
