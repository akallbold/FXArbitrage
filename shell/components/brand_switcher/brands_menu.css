@import 'playa-styleguide/src/stylesheets/variables.css';

:global(.is-open) .menu {
  display: block;
}

:root {
  --brand-menu-padding-lg: 12px;
}

.menu {
  display: none;
  position: absolute;
  width: 400px;
  max-height: calc(100vh - 170px);
  top: calc(100% + 10px);
  left: 5px;
  border-radius: 2px;
  border: 1px solid var(--grey);
  background-color: var(--white);
  box-shadow: var(--standard-shadow);
  z-index: 900;
}

.heading {
  padding: 10px var(--brand-menu-padding-lg) var(--brand-menu-padding-lg);
  border-bottom: solid 1px var(--grey);
  background-color: var(--background-grey);
}

.noResults {
  position: absolute;
  width: calc(100% - var(--brand-menu-padding-lg) * 2);
  top: 50%;
  transform: translateY(-50%);
  text-align: center;
}

.brandsList {
  position: relative;
  width: 100%;
  margin-bottom: 10px;
  padding: 0 var(--brand-menu-padding-lg) 20px;
  max-height: calc(100vh - 237px);
  min-height: 135px;
  overflow-x: hidden;
  overflow-y: scroll;
}
