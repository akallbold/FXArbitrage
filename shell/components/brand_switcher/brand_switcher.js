import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { get } from 'lodash'
import { DownArrowIcon, TinySpinner } from 'playa-styleguide'
import { brandsForSwitcher } from 'selectors/org_selectors'
import { openBrandSwitcher, closeBrandSwitcher } from './brand_switcher_actions.js'
import BrandsMenu from './brands_menu'
import Styles from './brand_switcher.css'

export class BrandSwitcher extends React.Component {
  constructor (props) {
    super(props)
    this.id = this.props.id || 'brand-switcher'
  }

  componentWillReceiveProps (nextProps) {
    if (!nextProps.isOpen && this.props.isOpen) {
      this.removeOutsideClickListener()
    } else if (nextProps.isOpen && !this.props.isOpen) {
      this.addOutsideClickListener()
    }
  }

  onClick = () => {
    if (this.props.isOpen) {
      this.props.closeBrandSwitcher()
    } else {
      this.props.openBrandSwitcher()
    }
  }

  outsideClickHandler = (event) => {
    if (document.getElementById(this.id).contains(event.target)) return false
    return this.props.closeBrandSwitcher()
  }

  addOutsideClickListener = () => window.addEventListener('click', this.outsideClickHandler)

  removeOutsideClickListener = () => window.removeEventListener('click', this.outsideClickHandler)

  render () {
    const { label, brands, isOpen } = this.props

    // get brand name from selected or first brand or null
    const brandName = get(brands.find(brand => brand.isSelected), 'name', brands[0] ? brands[0].name : null)
    const hasNoContent = !label && !brandName
    return (
      <div
        id='brand-switcher'
        className={`${Styles.switcher} ${isOpen ? 'is-open' : ''} ${hasNoContent ? 'no-content' : ''}`}
      >
        <span className={Styles.button}>
          <div className={Styles.secondaryText}>Brand</div>
          <div className={Styles.primaryText} onClick={this.onClick}>
            {!hasNoContent &&
              <span>{brandName || label} <DownArrowIcon className={Styles.downArrowIcon} /></span>
            }
            {hasNoContent &&
              <TinySpinner className={Styles.tinySpinner} />
            }
          </div>
        </span>
        {
          isOpen &&
          <BrandsMenu
            brands={brands}
          />
        }
      </div>
    )
  }
}

const mapStateToProps = state => ({
  brands: brandsForSwitcher(state),
  isOpen: state.brandSwitcher.isOpen
})

const mapDispatchToProps = {
  openBrandSwitcher,
  closeBrandSwitcher
}

BrandSwitcher.propTypes = {
  // just plain old props
  label: PropTypes.string,
  // state props
  brands: PropTypes.array.isRequired,
  isOpen: PropTypes.bool.isRequired,
  // dispatch props
  openBrandSwitcher: PropTypes.func.isRequired,
  closeBrandSwitcher: PropTypes.func.isRequired
}

BrandSwitcher.defaultProps = {
  isOpen: true,
  brands: [],
  label: '',
  startDate: '',
  endDate: ''
}

export default connect(mapStateToProps, mapDispatchToProps)(BrandSwitcher)
