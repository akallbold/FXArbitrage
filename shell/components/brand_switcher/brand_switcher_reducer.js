import { OPEN_BRAND_SWITCHER, CLOSE_BRAND_SWITCHER } from './brand_switcher_actions.js'

export const brandSwitcher = (state = { isOpen: false }, action) => {
  switch (action.type) {
    case OPEN_BRAND_SWITCHER: return { isOpen: true }
    case CLOSE_BRAND_SWITCHER: return { isOpen: false }
    default: return state
  }
}

export default brandSwitcher
