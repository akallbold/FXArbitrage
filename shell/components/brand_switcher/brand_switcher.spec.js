import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { BrandSwitcher } from './brand_switcher'
import {
  BrandsMenu,
  BrandsMenuList,
  createBrandUrl,
  formatListData,
  filterBrands
} from './brands_menu'
import { BrandMenuItem } from './brand_menu_item'

const nikeFootball = { id: 1, name: 'Nike/Football', companyId: 1, isSelected: true, companyName: 'Nike' }
const nikeBaseball = { id: 2, name: 'Nike/Baseball', companyId: 1, isSelected: false, companyName: 'Nike' }
const explora = { id: 3, name: 'Explora EasiFlow', companyId: 2, isSelected: false, companyName: 'Tommee Tippee' }
const sasquatch = { id: 4, name: 'Sasquatch', companyId: 3, isSelected: false, companyName: 'Baby Goat Wings' }
const sippee = { id: 5, name: 'Sippee Cups', companyId: 2, isSelected: false, companyName: 'Tommee Tippee' }
const furp = { id: 6, name: 'Furp', companyId: 3, isSelected: false, companyName: 'Baby Goat Wings' }
const babyGoatWings = { id: 7, name: 'Baby Goat Wings', companyId: 3, isSelected: false, companyName: 'Baby Goat Wings' }

const formattedBrands = [
  { label: 'Baby Goat Wings', isHeading: true, orgId: 3, orgType: 'company' },
  { label: 'Baby Goat Wings', hrefPathname: '/brands/7/facebook/manager/campaigns', isSelected: false, orgId: 7, orgType: 'brand' },
  { label: 'Furp', hrefPathname: '/brands/6/facebook/manager/campaigns', isSelected: false, orgId: 6, orgType: 'brand' },
  { label: 'Sasquatch', hrefPathname: '/brands/4/facebook/manager/campaigns', isSelected: false, orgId: 4, orgType: 'brand' },
  { label: 'Nike', isHeading: true, orgId: 1, orgType: 'company' },
  { label: 'Nike/Baseball', hrefPathname: '/brands/2/facebook/manager/campaigns', isSelected: false, orgId: 2, orgType: 'brand' },
  { label: 'Nike/Football', hrefPathname: '/brands/1/facebook/manager/campaigns', isSelected: true, orgId: 1, orgType: 'brand' },
  { label: 'Tommee Tippee', isHeading: true, orgId: 2, orgType: 'company' },
  { label: 'Explora EasiFlow', hrefPathname: '/brands/3/facebook/manager/campaigns', isSelected: false, orgId: 3, orgType: 'brand' },
  { label: 'Sippee Cups', hrefPathname: '/brands/5/facebook/manager/campaigns', isSelected: false, orgId: 5, orgType: 'brand' }
]

const brands = [
  { ...nikeFootball },
  { ...nikeBaseball },
  { ...explora },
  { ...sasquatch },
  { ...sippee },
  { ...furp },
  { ...babyGoatWings }
]

const isOpen = true
const openBrandSwitcher = () => {}
const closeBrandSwitcher = () => {}

describe('<BrandSwitcher />', () => {
  it('should render properly given all props', () => {
    const component = shallow(
      <BrandSwitcher {...{ brands, isOpen, openBrandSwitcher, closeBrandSwitcher }} />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render with `brandName` over `label` if available', () => {
    const component = shallow(
      <BrandSwitcher {...{ brands, label: 'Fallback Label', isOpen, openBrandSwitcher, closeBrandSwitcher }} />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render with fallback `label` given empty `brands`', () => {
    const component = shallow(
      <BrandSwitcher {...{ brands: [], label: 'Fallback Label', isOpen, openBrandSwitcher, closeBrandSwitcher }} />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render properly given empty `brands` AND no `label`', () => {
    const component = shallow(
      <BrandSwitcher {...{ brands: [], label: void 0, isOpen, openBrandSwitcher, closeBrandSwitcher }} />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render properly given `isOpen` is false', () => {
    const component = shallow(
      <BrandSwitcher {...{ brands, isOpen: false, openBrandSwitcher, closeBrandSwitcher }} />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})

describe('<BrandsMenu />', () => {
  const currentProps = {
    brands
  }

  const menu = (<BrandsMenu {...{ brands }} />)
  const component = shallow(menu)

  it('should render properly given all props', () => {
    expect(toJson(component)).toMatchSnapshot()
  })

  describe('componentWillReceiveProps', () => {
    it('should update if brands are not equal', () => {
      const component = shallow(menu)
      const inst = component.instance()
      const spy = jest.spyOn(inst, 'updateFilteredBrands')
      const nextNikeFootball = {
        ...nikeFootball,
        isSelected: false
      }
      const nextNikeBaseball = {
        ...nikeBaseball,
        isSelected: true
      }
      const nextBrands = [
        { ...nextNikeFootball },
        { ...nextNikeBaseball },
        { ...explora },
        { ...sasquatch },
        { ...sippee },
        { ...furp },
        { ...babyGoatWings }
      ]
      const nextProps = {
        ...currentProps,
        brands: nextBrands
      }
      inst.componentWillReceiveProps(nextProps)
      expect(spy).toHaveBeenCalled()
    })

    it('should NOT update if brands are the same', () => {
      const component = shallow(menu)
      const inst = component.instance()
      const spy = jest.spyOn(inst, 'updateFilteredBrands')
      const nextProps = { ...currentProps }
      inst.componentWillReceiveProps(nextProps)
      expect(spy).not.toHaveBeenCalled()
    })
  })
})

describe('filterBrands', () => {
  it('filterBrands should NOT mutate brands array', () => {
    filterBrands({ brands, query: 'nike' })
    expect(brands).toEqual(brands)
  })

  it('filterBrands should NOT mutate brands objects', () => {
    const filteredBrands = filterBrands({ brands, query: '' })
    filteredBrands[0].name = 'Mutated'
    expect(filteredBrands[0].name).not.toEqual(brands[0].name)
  })

  it('filterBrands should filter out irrelevant brands', () => {
    const expected = [{ ...nikeFootball }, { ...nikeBaseball }]
    const filteredBrands = filterBrands({ brands, query: 'nike' })
    expect(filteredBrands).toEqual(expected)
  })

  it('filterBrands should filter out no brands with \'\'', () => {
    const filteredBrands = filterBrands({ brands, query: '' })
    expect(filteredBrands).toEqual(brands)
  })

  it('filterBrands should filter out no brands with null', () => {
    const filteredBrands = filterBrands({ brands, query: null })
    expect(filteredBrands).toEqual(brands)
  })

  it('filterBrands should filter out no brands with undefined', () => {
    const filteredBrands = filterBrands({ brands, query: void 0 })
    expect(filteredBrands).toEqual(brands)
  })

  it('filterBrands should return empty Array when no matches', () => {
    const filteredBrands = filterBrands({ brands, query: 'wefwef' })
    expect(filteredBrands).toEqual([])
  })
})

describe('createBrandUrl', () => {
  it('should make a url based on active brand in pecking order', () => {
    const brandUrlFB = createBrandUrl({
      id: 149,
      advertisingState: 'ready',
      twAdvertisingStateReady: true,
      ptAdvertisingEnabled: false,
      scAdvertisingEnabled: false
    })
    expect(brandUrlFB.includes('/facebook')).toEqual(true)

    const brandUrlTW = createBrandUrl({
      id: 149,
      advertisingState: 'waiting_on_hyfn8',
      twAdvertisingStateReady: true,
      ptAdvertisingEnabled: false,
      scAdvertisingEnabled: false
    })
    expect(brandUrlTW.includes('/twitter')).toEqual(true)
  })

  it('should default to facebook if no active channel', () => {
    const brandUrlNone = createBrandUrl({
      id: 149,
      advertisingState: 'waiting_on_hyfn8',
      twAdvertisingStateReady: false,
      ptAdvertisingEnabled: false,
      scAdvertisingEnabled: false
    })
    expect(brandUrlNone.includes('/facebook')).toEqual(true)
  })
})

describe('formatListData', () => {
  it('formatListData should NOT mutate brands array', () => {
    formatListData(brands)
    expect(brands).toEqual(brands)
  })

  it('formatListData should NOT mutate brands objects', () => {
    const formattedList = formatListData(brands)
    formattedList[0].label = 'Mutated'
    expect(formattedList[0].label).not.toEqual(brands[0].companyName)
  })

  it('formatListData should format properly', () => {
    const formattedList = formatListData(brands)
    expect(formattedList).toEqual(formattedBrands)
  })
})

describe('<BrandsMenuList />', () => {
  it('should render properly given all props', () => {
    const component = shallow(
      <BrandsMenuList brands={formattedBrands} />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render no results message given empty brands', () => {
    const component = shallow(
      <BrandsMenuList brands={[]} query='testing' />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})

describe('<BrandMenuItem />', () => {
  it('should render a heading if `isHeading` is true', () => {
    const component = shallow(
      <BrandMenuItem {...formattedBrands[0]} />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render a menu item if `isHeading` is false', () => {
    const component = shallow(
      <BrandMenuItem {...formattedBrands[2]} />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render a menu item with selected class if `isSelected` is true', () => {
    const component = shallow(
      <BrandMenuItem {...formattedBrands[1]} />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})
