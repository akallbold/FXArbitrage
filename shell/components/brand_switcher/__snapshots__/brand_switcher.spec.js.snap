// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrandMenuItem /> should render a heading if \`isHeading\` is true 1`] = `
<div
  className="item heading"
>
  <Connect(Link)
    className="link"
    href={
      Object {
        "pathname": "/brands/[someBrandId]/[channel]/manager/",
        "query": Object {
          "refresh": 3,
        },
      }
    }
  >
    Baby Goat Wings
  </Connect(Link)>
</div>
`;

exports[`<BrandMenuItem /> should render a menu item if \`isHeading\` is false 1`] = `
<div
  className="item"
>
  <Connect(Link)
    className="link"
    href={
      Object {
        "pathname": "/brands/6/facebook/manager/campaigns",
        "query": Object {
          "refresh": 6,
        },
      }
    }
  >
    Furp
  </Connect(Link)>
</div>
`;

exports[`<BrandMenuItem /> should render a menu item with selected class if \`isSelected\` is true 1`] = `
<div
  className="item"
>
  <Connect(Link)
    className="link"
    href={
      Object {
        "pathname": "/brands/7/facebook/manager/campaigns",
        "query": Object {
          "refresh": 7,
        },
      }
    }
  >
    Baby Goat Wings
  </Connect(Link)>
</div>
`;

exports[`<BrandSwitcher /> should render properly given \`isOpen\` is false 1`] = `
<div
  className="switcher  "
  id="brand-switcher"
>
  <span
    className="button"
  >
    <div
      className="secondaryText"
    >
      Brand
    </div>
    <div
      className="primaryText"
      onClick={[Function]}
    >
      <span>
        Nike/Football
         
        <test-file-stub
          className="downArrowIcon"
        />
      </span>
    </div>
  </span>
</div>
`;

exports[`<BrandSwitcher /> should render properly given all props 1`] = `
<div
  className="switcher is-open "
  id="brand-switcher"
>
  <span
    className="button"
  >
    <div
      className="secondaryText"
    >
      Brand
    </div>
    <div
      className="primaryText"
      onClick={[Function]}
    >
      <span>
        Nike/Football
         
        <test-file-stub
          className="downArrowIcon"
        />
      </span>
    </div>
  </span>
  <Connect(BrandsMenu)
    brands={
      Array [
        Object {
          "companyId": 1,
          "companyName": "Nike",
          "id": 1,
          "isSelected": true,
          "name": "Nike/Football",
        },
        Object {
          "companyId": 1,
          "companyName": "Nike",
          "id": 2,
          "isSelected": false,
          "name": "Nike/Baseball",
        },
        Object {
          "companyId": 2,
          "companyName": "Tommee Tippee",
          "id": 3,
          "isSelected": false,
          "name": "Explora EasiFlow",
        },
        Object {
          "companyId": 3,
          "companyName": "Baby Goat Wings",
          "id": 4,
          "isSelected": false,
          "name": "Sasquatch",
        },
        Object {
          "companyId": 2,
          "companyName": "Tommee Tippee",
          "id": 5,
          "isSelected": false,
          "name": "Sippee Cups",
        },
        Object {
          "companyId": 3,
          "companyName": "Baby Goat Wings",
          "id": 6,
          "isSelected": false,
          "name": "Furp",
        },
        Object {
          "companyId": 3,
          "companyName": "Baby Goat Wings",
          "id": 7,
          "isSelected": false,
          "name": "Baby Goat Wings",
        },
      ]
    }
  />
</div>
`;

exports[`<BrandSwitcher /> should render properly given empty \`brands\` AND no \`label\` 1`] = `
<div
  className="switcher is-open no-content"
  id="brand-switcher"
>
  <span
    className="button"
  >
    <div
      className="secondaryText"
    >
      Brand
    </div>
    <div
      className="primaryText"
      onClick={[Function]}
    >
      <test-file-stub
        className="tinySpinner"
      />
    </div>
  </span>
  <Connect(BrandsMenu)
    brands={Array []}
  />
</div>
`;

exports[`<BrandSwitcher /> should render with \`brandName\` over \`label\` if available 1`] = `
<div
  className="switcher is-open "
  id="brand-switcher"
>
  <span
    className="button"
  >
    <div
      className="secondaryText"
    >
      Brand
    </div>
    <div
      className="primaryText"
      onClick={[Function]}
    >
      <span>
        Nike/Football
         
        <test-file-stub
          className="downArrowIcon"
        />
      </span>
    </div>
  </span>
  <Connect(BrandsMenu)
    brands={
      Array [
        Object {
          "companyId": 1,
          "companyName": "Nike",
          "id": 1,
          "isSelected": true,
          "name": "Nike/Football",
        },
        Object {
          "companyId": 1,
          "companyName": "Nike",
          "id": 2,
          "isSelected": false,
          "name": "Nike/Baseball",
        },
        Object {
          "companyId": 2,
          "companyName": "Tommee Tippee",
          "id": 3,
          "isSelected": false,
          "name": "Explora EasiFlow",
        },
        Object {
          "companyId": 3,
          "companyName": "Baby Goat Wings",
          "id": 4,
          "isSelected": false,
          "name": "Sasquatch",
        },
        Object {
          "companyId": 2,
          "companyName": "Tommee Tippee",
          "id": 5,
          "isSelected": false,
          "name": "Sippee Cups",
        },
        Object {
          "companyId": 3,
          "companyName": "Baby Goat Wings",
          "id": 6,
          "isSelected": false,
          "name": "Furp",
        },
        Object {
          "companyId": 3,
          "companyName": "Baby Goat Wings",
          "id": 7,
          "isSelected": false,
          "name": "Baby Goat Wings",
        },
      ]
    }
  />
</div>
`;

exports[`<BrandSwitcher /> should render with fallback \`label\` given empty \`brands\` 1`] = `
<div
  className="switcher is-open "
  id="brand-switcher"
>
  <span
    className="button"
  >
    <div
      className="secondaryText"
    >
      Brand
    </div>
    <div
      className="primaryText"
      onClick={[Function]}
    >
      <span>
        Fallback Label
         
        <test-file-stub
          className="downArrowIcon"
        />
      </span>
    </div>
  </span>
  <Connect(BrandsMenu)
    brands={Array []}
  />
</div>
`;

exports[`<BrandsMenu /> should render properly given all props 1`] = `
<div
  className="menu"
  onKeyDown={[Function]}
>
  <div
    className="heading"
  >
    <SearchInput
      className="searchInput"
      inputProps={
        Object {
          "placeholder": "Find Brands",
        }
      }
      isLive={true}
      name="filter-brands"
      onChange={[Function]}
      submitIcon={null}
    />
  </div>
  <BrandsMenuList
    arrowKeyedToIndex={0}
    brands={
      Array [
        Object {
          "isHeading": true,
          "label": "Baby Goat Wings",
          "orgId": 3,
          "orgType": "company",
        },
        Object {
          "hrefPathname": "/brands/7/facebook/manager/campaigns",
          "isSelected": false,
          "label": "Baby Goat Wings",
          "orgId": 7,
          "orgType": "brand",
        },
        Object {
          "hrefPathname": "/brands/6/facebook/manager/campaigns",
          "isSelected": false,
          "label": "Furp",
          "orgId": 6,
          "orgType": "brand",
        },
        Object {
          "hrefPathname": "/brands/4/facebook/manager/campaigns",
          "isSelected": false,
          "label": "Sasquatch",
          "orgId": 4,
          "orgType": "brand",
        },
        Object {
          "isHeading": true,
          "label": "Nike",
          "orgId": 1,
          "orgType": "company",
        },
        Object {
          "hrefPathname": "/brands/2/facebook/manager/campaigns",
          "isSelected": false,
          "label": "Nike/Baseball",
          "orgId": 2,
          "orgType": "brand",
        },
        Object {
          "hrefPathname": "/brands/1/facebook/manager/campaigns",
          "isSelected": true,
          "label": "Nike/Football",
          "orgId": 1,
          "orgType": "brand",
        },
        Object {
          "isHeading": true,
          "label": "Tommee Tippee",
          "orgId": 2,
          "orgType": "company",
        },
        Object {
          "hrefPathname": "/brands/3/facebook/manager/campaigns",
          "isSelected": false,
          "label": "Explora EasiFlow",
          "orgId": 3,
          "orgType": "brand",
        },
        Object {
          "hrefPathname": "/brands/5/facebook/manager/campaigns",
          "isSelected": false,
          "label": "Sippee Cups",
          "orgId": 5,
          "orgType": "brand",
        },
      ]
    }
    query=""
  />
</div>
`;

exports[`<BrandsMenuList /> should render no results message given empty brands 1`] = `
<div
  className="brandsList"
>
  <h4
    className="noResults"
  >
    Sorry, no results for "
    testing
    "
  </h4>
</div>
`;

exports[`<BrandsMenuList /> should render properly given all props 1`] = `
<div
  className="brandsList"
>
  <BrandMenuItem
    arrowKeyedTo={true}
    classes=""
    hrefPathname="/brands/[someBrandId]/[channel]/manager/"
    isHeading={true}
    isSelected={false}
    itemRef={[Function]}
    label="Baby Goat Wings"
    orgId={3}
    orgType="company"
  />
  <BrandMenuItem
    arrowKeyedTo={false}
    classes=""
    hrefPathname="/brands/7/facebook/manager/campaigns"
    isHeading={false}
    isSelected={false}
    itemRef={[Function]}
    label="Baby Goat Wings"
    orgId={7}
    orgType="brand"
  />
  <BrandMenuItem
    arrowKeyedTo={false}
    classes=""
    hrefPathname="/brands/6/facebook/manager/campaigns"
    isHeading={false}
    isSelected={false}
    itemRef={[Function]}
    label="Furp"
    orgId={6}
    orgType="brand"
  />
  <BrandMenuItem
    arrowKeyedTo={false}
    classes=""
    hrefPathname="/brands/4/facebook/manager/campaigns"
    isHeading={false}
    isSelected={false}
    itemRef={[Function]}
    label="Sasquatch"
    orgId={4}
    orgType="brand"
  />
  <BrandMenuItem
    arrowKeyedTo={false}
    classes=""
    hrefPathname="/brands/[someBrandId]/[channel]/manager/"
    isHeading={true}
    isSelected={false}
    itemRef={[Function]}
    label="Nike"
    orgId={1}
    orgType="company"
  />
  <BrandMenuItem
    arrowKeyedTo={false}
    classes=""
    hrefPathname="/brands/2/facebook/manager/campaigns"
    isHeading={false}
    isSelected={false}
    itemRef={[Function]}
    label="Nike/Baseball"
    orgId={2}
    orgType="brand"
  />
  <BrandMenuItem
    arrowKeyedTo={false}
    classes=""
    hrefPathname="/brands/1/facebook/manager/campaigns"
    isHeading={false}
    isSelected={true}
    itemRef={[Function]}
    label="Nike/Football"
    orgId={1}
    orgType="brand"
  />
  <BrandMenuItem
    arrowKeyedTo={false}
    classes=""
    hrefPathname="/brands/[someBrandId]/[channel]/manager/"
    isHeading={true}
    isSelected={false}
    itemRef={[Function]}
    label="Tommee Tippee"
    orgId={2}
    orgType="company"
  />
  <BrandMenuItem
    arrowKeyedTo={false}
    classes=""
    hrefPathname="/brands/3/facebook/manager/campaigns"
    isHeading={false}
    isSelected={false}
    itemRef={[Function]}
    label="Explora EasiFlow"
    orgId={3}
    orgType="brand"
  />
  <BrandMenuItem
    arrowKeyedTo={false}
    classes=""
    hrefPathname="/brands/5/facebook/manager/campaigns"
    isHeading={false}
    isSelected={false}
    itemRef={[Function]}
    label="Sippee Cups"
    orgId={5}
    orgType="brand"
  />
</div>
`;
